flink-streaming-java/src/test/java/org/apache/flink/streaming/runtime/tasks/StreamTaskTest.java

```diff
@@ -1763,6 +1765,108 @@ public long calculateThroughput() {
         }
     }
 
+    /**
+     * Tests mailbox metrics latency and queue size and verifies that (1) latency measurement is
+     * executed initially once and at least once triggered by timer, (2) latency max value is
+     * greater than zero and (3) mailbox size is greater than zero for some time and eventually
+     * equals to zero.
+     *
+     * <p>Note: This test uses a timeout which, if exceeded, indicates that the minimum number of
+     * latency measurements has not reached the specified minimum within specified time.
+     *
+     * @throws Exception on {@link MockEnvironmentBuilder#build()} failure.
+     */
+    @Test(timeout = 4000)
```

Comment:
I would remove this timeout. Our general practise is to avoid timeouts inside the tests. If there is a deadlock/livelock, it's actually better for mvn watchdog to detect that, and kill the build with printing thread dump.

Apart of that 4s is most likely too short. We have seen occasional 10-15sec freezes in the CI.